src/Symfony/Component/VarDumper/Caster/SocketCaster.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\VarDumper\Caster;
+
+use Symfony\Component\VarDumper\Cloner\Stub;
+
+/**
+ * @author Nicolas Grekas <p@tchwork.com>
+ */
+final class SocketCaster
+{
+    public static function castSocket(\Socket $h, array $a, Stub $stub, bool $isNested): array
+    {
+        if (socket_atmark($h)) {
+            $a[Caster::PREFIX_VIRTUAL.'atmark'] = true;
+        }
+
+        if (!$lastError = socket_last_error($h)) {
+            return $a;
+        }
+
+        static $errors;
+
+        if (!$errors) {
+            $errors = get_defined_constants(true)['sockets'] ?? [];
+            $errors = array_flip(array_filter($errors, static fn ($k) => str_starts_with($k, 'SOCKET_E'), \ARRAY_FILTER_USE_KEY));
+        }
+
+        $a[Caster::PREFIX_VIRTUAL.'last_error'] = new ConstStub($errors[$lastError], socket_strerror($lastError));
```

Comment:
Do we have a guarantee that the last error will be in `$errors` or should we handle that case gracefully to avoid issues (getting an error triggered by the debugging tool would be painful for users) ?

Reply:
it'd say yes, it cannot be otherwise